### ISSUE TITLE
Check that the smoothest path's Wald critical value is correct

### ISSUE THREAD

**Issue by Constantino-Carreto-Romero** [JMSLab/xtevent#131]
per call: check that the smoothest path's Wald critical value is correct

**Comment by jorpppp** [comment]
It seems like we have the wrong degrees of freedom for the Wald critical value, as noted here by @SimonFreyaldenhoven @santiagohermo

https://github.com/JMSLab/eventstudyr/issues/19#issuecomment-1440803153

We need to subtract the number of normalized coefficients from the degrees of freedom.

**Comment by jorpppp** [comment]
@santiagohermo I am testing the fix for this issue. Can you please confirm that the data you were using for the tests here https://github.com/JMSLab/eventstudyr/issues/19#issuecomment-1440803153  comes from this file, please?:

https://github.com/JMSLab/eventstudyr/blob/ce54435e869a1198c2a41d83c3c4e5c515bf3154/examples/source/raw/eventstudy_illustration_data/orig/simulation_data_dynamic.dta

If not, can you post a link to the test data you were using?

**Comment by jmshapir** [comment]
@santiagohermo would you be able to post a self-contained small ZIP archive with whatever data or code @jorpppp will need to reproduce the issue? That way we won't need to rely on links to an external repository here.

Thanks!

**Comment by santiagohermo** [comment]
Thanks @jorpppp @jmshapir! To reply to @jorpppp, the answer is yes. The data ultimately comes from the the [data archive of the paper](https://data.nber.org/data-appendix/w29170/). The exact version I used is in `.Rdata` format and can be accessed in [df_sample_dynamic.zip](https://github.com/JMSLab/xtevent/files/10924234/df_sample_dynamic.zip). I transformed it into csv here: [df_sample_dynamic_csv.zip](https://github.com/JMSLab/xtevent/files/10924255/df_sample_dynamic_csv.zip).

To conduct the analysis we discussed in the call I made some changes to my local `xteventplot.ado`:
1. I dropped the `cap qui` in [this line](https://github.com/JMSLab/xtevent/blob/aff9157385029ff1a8f229274086b687cd342f26/xtevent/xteventplot.ado#L445)
2. I added the following lines right after the line above:
 
  ```stata
  mata: st_matrix("param",param)
  di _n "Smothest path:"
  mat list param

  mata: st_matrix("WB",WB)
  di _n "Optimal Wald value"
  mat list WB
  ```
3. Then I ran stata from the cmd to get a log file with `WB` values.

Let me know if I can be of any more help @jorpppp!

**Comment by jorpppp** [comment]
In https://github.com/JMSLab/xtevent/commit/e7056c4d6087a3fb496c3ec86e3535e01e33feb9 and https://github.com/JMSLab/xtevent/commit/fff916b616293a37d0c81456cf17f4219aa01792 I fixed the critical value and I can confirm that now the critical value is the one corresponding to the number of estimated parameters (omitting the normalized parameters).

I also added mata saves for the Wald critical value and the Wald optimal value in case we need further debugging later. I'll start a PR now.
